apps/worker/src/app/health/health.controller.ts

```diff
@@ -1,40 +1,26 @@
-import { Controller, Get } from '@nestjs/common';
+import { Controller, Get, Inject } from '@nestjs/common';
 import { ApiExcludeController } from '@nestjs/swagger';
 import { HealthCheck, HealthCheckResult, HealthCheckService } from '@nestjs/terminus';
-import {
-  DalServiceHealthIndicator,
-  StandardQueueServiceHealthIndicator,
-  WorkflowQueueServiceHealthIndicator,
-  ActiveJobsMetricQueueServiceHealthIndicator,
-  CompletedJobsMetricQueueServiceHealthIndicator,
-  SubscriberProcessQueueHealthIndicator,
-} from '@novu/application-generic';
+import { DalServiceHealthIndicator, QueueHealthIndicator } from '@novu/application-generic';
 
 import { version } from '../../../package.json';
 
 @Controller('health-check')
 @ApiExcludeController()
 export class HealthController {
   constructor(
+    @Inject('QUEUE_HEALTH_INDICATORS')
+    private healthIndicators: QueueHealthIndicator[],
     private healthCheckService: HealthCheckService,
-    private dalHealthIndicator: DalServiceHealthIndicator,
-    private standardQueueHealthIndicator: StandardQueueServiceHealthIndicator,
-    private workflowQueueHealthIndicator: WorkflowQueueServiceHealthIndicator,
-    private activeJobsMetricQueueServiceHealthIndicator: ActiveJobsMetricQueueServiceHealthIndicator,
-    private completedJobsMetricQueueServiceHealthIndicator: CompletedJobsMetricQueueServiceHealthIndicator,
-    private subscriberProcessQueueHealthIndicator: SubscriberProcessQueueHealthIndicator
+    private dalHealthIndicator: DalServiceHealthIndicator
   ) {}
 
   @Get()
   @HealthCheck()
   healthCheck(): Promise<HealthCheckResult> {
     return this.healthCheckService.check([
       async () => this.dalHealthIndicator.isHealthy(),
-      async () => this.standardQueueHealthIndicator.isActive(),
-      async () => this.workflowQueueHealthIndicator.isActive(),
-      async () => this.activeJobsMetricQueueServiceHealthIndicator.isActive(),
-      async () => this.completedJobsMetricQueueServiceHealthIndicator.isActive(),
-      async () => this.subscriberProcessQueueHealthIndicator.isActive(),
+      ...this.healthIndicators.map((indicator) => async () => indicator.isHealthy()),
```

Comment:
We now dynamically generated this list depending on the worker context of active workers